docs/database/cosmosdb.md

```diff
@@ -61,9 +61,10 @@ For batch operations, if the individual operations are known to have the same op
 
 **[7]:** If using a port other than the default port for this DBMS and if `server.address` is set.
 
-**[8]:** For batch operations, if the individual operations are known to have the same query text then that query text SHOULD be used, otherwise all of the individual query texts SHOULD be concatenated with separator `; ` or some other database system specific separator if more applicable.
+**[8]:** Even though parameterized query text can potentially have sensitive data, by using a parameterized query the user is giving a strong signal that any sensitive data will be passed as parameter values, and the benefit to observability of capturing the static part of the query text by default outweighs the risk.
 
-**[9]:** SHOULD be collected by default only if there is sanitization that excludes sensitive information.
+**[9]:** Non-parameterized query text SHOULD NOT be collected by default unless there is sanitization that excludes sensitive data. Sanitization SHOULD be done by replacing all literals with a placeholder value. Such literals include, but are not limited to, String literals, Numeric literals, Date and Time literals, Boolean literals, Interval literals, Binary literals, and Hexadecimal literals. The placeholder value SHOULD be `?`, unless it already has a defined meaning in the given database system, in which case the instrumentation MAY choose a different placeholder.
```

Comment:
I think this text belongs on the attribute note (and not on the requirement level) since it talks about the format, not the condition under which the attribute should be collected. 
Also examples in the second sentence are definitely related to the existing parametrization note.

Reply:
I wonder if the rules/format of how to sanitize should be it's own section (or maybe file), since can be complex. This way we can add examples, considerations, etc, and then all places can just point to that link.

Then I can add the rules of grouping variables (issue #1053) on the same place.

Reply:
Yeah, totally agree, we should move this.

> should be it's own section (or maybe file)

So where should we move it? The big advantage of having this in yaml is that it'll be always auto generated (like this md file) - the downside is that the best we have there is to put this into a `note`, right? 

If we move it into a file, then we'll need to manually maintain it.

What should we do?

Reply:
From June 5 SIG call: Is there any way to determine whether a query has been sanitized? `?` is a common placeholder for query parameters.

An explicit flag that a query was sanitized could work, with the benefit of not require any string parsing to determine if the query was originally parameterized or not.

Reply:
> If we move it into a file, then we'll need to manually maintain it.

I see this as the equivalent of manually maintaining the yaml file. If there is a new md file, your yaml could have the note pointing to it, so that part continues to be auto generated.

The new file could be on `docs/database/sanitization.md`

Reply:
Discussed in the WG meeting - we add it to `database-spans.md` for now. Happy to move it out later if needed. We link to that section from yaml.

Reply:
Awesome, I like how it looks now! Thank you for making the update!

Reply:
> From June 5 SIG call: Is there any way to determine whether a query has been sanitized? `?` is a common placeholder for query parameters.
> 
> An explicit flag that a query was sanitized could work, with the benefit of not require any string parsing to determine if the query was originally parameterized or not.

That's a good point, however I'm not sure if this is important enough to address it at this point. We can always add an additional attribute to store if the query was sanitized or not as an incremental change.

There is the `db.query.parameter.<key>` to store parameters - so if the query is parametrized, then that attributes is used - however that attribute is opt-in, so admittedly that'll not help in a default setup.

I personally would wait for user feedback on this and if needed, then I'd add the additional flag on-demand if we really see the need for it.

Reply:
> I personally would wait for user feedback on this and if needed, then I'd add the additional flag on-demand if we really see the need for it.

let's create an issue for this 👍

Reply:
Created https://github.com/open-telemetry/semantic-conventions/issues/1179